### PR TITLE
SDK-919 CanHandle

### PR DESCRIPTION
**Pull request by jhaven-stytch**
Linear Ticket: [SDK-919](https://linear.app/stytch/issue/SDK-919)

## Changes:

1. Add canHandle method to clients
2. Split out token types so each client type only returns for their supported tokens
3. Add/update tests

## Notes:
- This is a little bigger than the iOS implementation, because I chose to split out the unique token types between consumer and B2B, so that if a user tries to validate a consumer deep link with a B2B client, it returns as expected. This might be something you want to address in iOS (it's edge-casey, but figured I'd call it out).

## Checklist:
- [x] I have verified that this change works in the relevant demo app, or N/A
- [x] I have added or updated any tests relevant to this change, or N/A
- [x] I have updated any relevant README files for this change, or N/A

### CODE REVIEW

File: sdk/src/main/java/com/stytch/sdk/b2b/B2BTokenType.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    MULTI_TENANT_MAGIC_LINKS,
+
+    /**
+     * An unknown deeplink type. It's possible a non-Stytch deeplink was supplied to the Stytch client's handle() method
+     */
+    UNKNOWN;
+
+    internal companion object {
+        fun fromString(typeString: String?): B2BTokenType {
```
Comment (dan-stytch) [inline sdk/src/main/java/com/stytch/sdk/b2b/B2BTokenType.kt:21]: Nit/Q: Could this function be a generic function on TokenType that returns a `Self` or something, so the implementation could be shared vs duplicated?

**Review by dan-stytch — APPROVED** [review]
🚢

File: sdk/src/main/java/com/stytch/sdk/b2b/B2BTokenType.kt (hunk 1 above)
Comment (jhaven-stytch) [reply sdk/src/main/java/com/stytch/sdk/b2b/B2BTokenType.kt:21]: Unfortunately, no. The interface has no knowledge of the implementing types, so can't check the values, and enum classes can't be extended :/
